### PR TITLE
AYS-246 | Admin Registration Flows Have Been Refactored and New Data Model Has Been Integrated

### PR DESCRIPTION
**Pull request by agitrubard**
### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

**Review by Furkan-Ahmet-Ozdemir — APPROVED** [review]
Good

**Review by MenekseYuncu — DISMISSED** [review]
Eline sağlık

**Review by MenekseYuncu — DISMISSED** [review]

File: src/main/java/org/ays/institution/controller/InstitutionController.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ class InstitutionController {
      * @return An {@link AysResponse} containing a list of {@link InstitutionResponse} representing the summary of institutions.
      */
     @GetMapping("/institutions/summary")
-    @PreAuthorize("hasAnyAuthority('SUPER_ADMIN')")
+    @PreAuthorize("hasAnyAuthority('SUPER_ADMIN', 'application:registration:create')")
```
Comment (moaydogdu) [inline src/main/java/org/ays/institution/controller/InstitutionController.java:40]: 'SUPER_ADMIN' i kullanmıyorsak kaldıralım unutmayalım.

**Review by moaydogdu — APPROVED** [review]

File: src/main/java/org/ays/institution/controller/InstitutionController.java (hunk 1 above)
Comment (agitrubard) [reply src/main/java/org/ays/institution/controller/InstitutionController.java]: Fixed ✅

**Review by skayikci — APPROVED** [review]

**Comment by skayikci** [comment]
TESEKKURLER!

**Review by MenekseYuncu — APPROVED** [review]

**Pull request merged**